### PR TITLE
doc: editor pass on release notes

### PR DESCRIPTION
**Pull request by dbkinder**
Do a review and editor pass on the release notes, reduce length of some
lists by using hlist extension, tweak css for appearance of hlist in a
bullet list.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/releases/release-notes-2.0.rst [hunk 1]
```diff
@@ -356,44 +368,39 @@ Networking
...
 *********
 
 * Host:
 
-  * GATT: Added support for database hashes
-  * GATT: Added support for Ready Using Characteristic UUID
-  * GATT: Added support for static services
-  * GATT: Added support for disabling the dynamic database
-  * GATT: Added support for notifying and indicating by UUID
+  * GATT: Added support for database hashes, Ready Using Characteristic
```
Comment (carlescufi) [inline doc/releases/release-notes-2.0.rst]: s/Ready/Read
(this was my mistake originally)

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by aescolar — APPROVED** [review]
Just a very minor further request.
The changes look nice to me

**Review by jukkar — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by ioannisg** [comment]
Could you rebase this PR @dbkinder ?

File: doc/releases/release-notes-2.0.rst [hunk 2]
```diff
@@ -77,67 +77,79 @@ Boards & SoC Support
 ********************
 
 * Add native_posix_64: A 64 bit variant of native_posix
```
[the remark this answers is not in the document]
Comment (dbkinder) [reply doc/releases/release-notes-2.0.rst]: Yep, good catch.

**Review by carlescufi — APPROVED** [review]

**Comment by ioannisg** [comment]
@dbkinder this, now, needs one more rebase :)

**Comment by ioannisg** [comment]
the reason is the PR with the HAL text that I merged right now :)

**Comment by dbkinder** [comment]
@ioannisg :)

**Review by ioannisg — APPROVED** [review]

**Pull request merged**